deepchem/feat/molecule_featurizers/mxmnet_featurizer.py

```diff
@@ -0,0 +1,154 @@
+
+    This featurizer is based on
+    `Molecular Mechanics-Driven Graph Neural Network with Multiplex Graph for Molecular Structures <https://arxiv.org/pdf/2011.07457.pdf>`_.
+
+    Examples
+    --------
+    >>> smiles = ["C1=CC=CN=C1", "C1CCC1"]
+    >>> featurizer = MXMNetFeaturizer()
+    >>> out = featurizer.featurize(smiles)
+    >>> type(out[0])
+    <class 'deepchem.feat.graph_data.GraphData'>
+    >>> out[0].num_nodes
+    6
+    >>> out[0].num_node_features
+    1
+    >>> out[0].node_features.shape
+    (6, 1)
+    >>> out[0].num_edges
+    12
+
+    Note
+    ----
+    We are not explitly handling hydrogen atoms for now. We only support 'H', 'C', 'N', 'O' and 'F' atoms to be present in the smiles at this point for MXMNet Model.
+
+    """
+
+    def __init__(self):
+
+        super().__init__()
+
+    def _construct_bond_index(self, datapoint: RDKitMol) -> np.ndarray:
+        """
+        Construct edge (bond) index
+
+        Parameters
+        ----------
+        datapoint: RDKitMol
+            RDKit mol object.
+
+        Returns
+        -------
+        edge_index: np.ndarray
+            Edge (Bond) index
+
+        """
+        edge_type: List = []
+        N: int = datapoint.GetNumAtoms()
+
+        # row, col = edge_index
+        src: List[int] = []
+        dest: List[int] = []
+        for bond in datapoint.GetBonds():
+            start, end = bond.GetBeginAtomIdx(), bond.GetEndAtomIdx()
+            src += [start, end]
+            dest += [end, start]
+            edge_type += 2 * [bonds[bond.GetBondType()]]
+
+        edge_index: torch.Tensor = torch.tensor([src, dest], dtype=torch.long)
+
+        perm: torch.Tensor = (edge_index[0] * N + edge_index[1]).argsort()
+        edge_index = edge_index[:, perm]
+
+        return np.asarray([src, dest], dtype=int)
+
+    def _featurize(self, datapoint: RDKitMol, **kwargs) -> GraphData:
+        """Calculate molecule graph features from RDKit mol object.
+
+        Parameters
+        ----------
+        datapoint: RDKitMol
+            RDKit mol object.
+
+        Returns
+        -------
+        graph: GraphData
+            A molecule graph object with features:
+            - node_features: np.ndarray
+                    Node feature matrix with shape [num_nodes, num_node_features]
+            - edge_index: np.ndarray, dtype int
+                    Graph connectivity in COO format with shape [2, num_edges]
+            - node_pos_features: np.ndarray, optional (default None)
+                    Node position matrix with shape [num_nodes, num_dimensions].
+        """
+        if not isinstance(datapoint, Chem.rdchem.Mol):
+            raise ValueError(
+                "Feature field should contain smiles for DMPNN featurizer!")
+
+        pos: List = []
+
+        # load_sdf_files returns pos as strings but user can also specify
+        # numpy arrays for atom coordinates
+        if 'pox_x' in kwargs and 'pos_y' in kwargs and 'posz' in kwargs:
```

Comment:
There is a typo in 'posz' variable. Please write a unit test for this functionality.

Reply:
Yes corrected, Thank you so much.